rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,602 @@
+        molViewState  = globals()['mol_views'][uid]
+        
+        
+        # handling mol and conf
+        molViewState = handleMolAndConf(molViewState)
+        
+        # handling properties
+        handleProperty(molViewState)
+        
+        # Add models (molecules/conformations) to viewer
+        pModelNum = 0
+        for model in molViewState.selectedModels:
+            pModelNum = pModelNum+1
+            globals()['view_'+uid].addModel(model, 'sdf')
+            
+        # Ligand Drawing style and colors
+        view=handleLigandStyle(uid, view)
+        # Ligand labeling
+        view = handleLigandLabeling(molViewState, view)
+        
+        # Add model (protein) to viewer
+        if 'protein_'+uid in globals():
+            # Show protein if visibitity is true
+            if globals()['proteinVisible_'+uid].value:
+                pdb = Chem.MolToPDBBlock(globals()['protein_'+uid])
+                view.addModel(pdb,'pdb')
+                # Protein Drawing style and colors
+                view = handleProteinStyle(uid, view, pModelNum)
+                
+        # zoomTo does not work well for surface and label... so, zoomTo should not be default settings
+        #view.zoomTo()
+        view.setBackgroundColor(globals()['background_'+uid].value)
+        
+        display(view.update())
+        
+    
+def handle_change(change):
+    """This function handles all the interactive widgets except buttons and 3Dmol.js viewer"""
+    update3D(change.owner._model_id)
+    
+def handle_start_button(b):
+    """This function handles start button."""
+    b.icon='check'
+    b.description="Done!"
+    uid=globals()['ipy_wgs'][b._model_id]
+    globals()['rdkit_3dviewer_start_button_clicked_'+uid] = True
+    update3D(b._model_id)
+    
+def handle_zoomTo_button(b):
+    """This function handles zoomTo button"""
+    uid=globals()['ipy_wgs'][b._model_id]
+    globals()['view_'+uid].zoomTo()
+    display(globals()['view_'+uid].update())
+    
+    
+def ChangeActiveLigand(uid, molId, confId, keepExistingModels = True):
+    """This function handles ligand select through python code
+    uid : string 
+    molId : string 
+    confId : intiger
+    keepExistingModels : logical (if False - all models are removed - molId and confId associated model rendered)
+    """
+    if type(uid) != 'str':
+        uid = str(uid)
+    if type(molId) != 'str':
+        molId = str(molId)
+        
+    if globals()['molId_'+uid].value != molId:
+        globals()['molId_'+uid].value = molId
+        
+    if globals()['confId_'+uid].value != confId:
+        globals()['confId_'+uid].value = confId
+    
+    if keepExistingModels is False:
+        globals()['selectMultiMols_'+uid].value=False
+        globals()['selectAllMols_'+uid].value=False
+        globals()['selectMultiConfs_'+uid].value=False
+        globals()['selectAllConfs_'+uid].value=False
+    
+    update3D(globals()['molId_'+uid]._model_id)
```

Comment:
You use the `uid` to create a combined key for retrieving information from globals(). That's there are all these statements like:
```
globals()['specifier_'+uid]...
```
This is a very unstructured way of storing the data. I would suggest the following:
```
globals()['viewData'] = {}
globals()['viewData'][uid] = {}
```
and use the dictionary `globals()['viewData'][uid]` to manage all the data for one of the molecule viewers. The above function is then:
```
def changeActiveLigand(uid, molId, confId, keepExistingModels=True):
    viewData = globals()['viewData'][uid]
    if viewData['molId'].value != molId:
        viewData['molId'].value = molId
    if viewData['confId'].value != confId:
        viewData['confId'].value = confId

    if not keepExistingModels:
        viewData['selectMultiMols'].value = False
        viewData['selectAllMols'].value = False
        viewData['selectMultiConfs'].value = False
        viewData['selectAllConfs'].value = False

  update3D(viewData['molId']._model_id)
```
You can even replace `uid` with `viewData` if that is available in the calling function.

In general, it is seen as bad practice to access 'hidden' properties (indicated by name starting with `_`). I assume there is a good reason here though.